File: tests/src/org.apache.calcite/calcite-core/1.35.0/src/test/java/org_apache_calcite/calcite_core/ResourcesTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_apache_calcite.calcite_core;

import org.apache.calcite.runtime.CalciteResource;
import org.apache.calcite.runtime.Resources;
import org.junit.jupiter.api.Test;

import java.util.EnumSet;

import static org.assertj.core.api.Assertions.assertThat;

public class ResourcesTest {
    @Test
    void createsAndValidatesResourceProxy() {
        CalciteResource resource = Resources.create(CalciteResource.class);

        assertThat(resource.badFormat("date").str()).contains("date");
        Resources.validate(resource, EnumSet.of(
                Resources.Validation.AT_LEAST_ONE,
                Resources.Validation.MESSAGE_SPECIFIED));
    }
}

File: tests/src/org.scalactic/scalactic_3/3.2.19/src/test/scala/org_scalactic/scalactic_3/ResourcesTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package org_scalactic.scalactic_3

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import org.scalactic.Requirements

class ResourcesTest {
  @Test
  def failedRequirementLoadsLocalizedFailureMessage(): Unit = {
    val thrown: IllegalArgumentException = assertThrows(
      classOf[IllegalArgumentException],
      () => Requirements.require(false)
    )

    assertThat(thrown.getMessage).isNotBlank()
    assertThat(thrown.getMessage).contains("false")
  }
}
